tests/shared/src/test/scala/scala/meta/tests/parsers/ScaladocParserSuite.scala

```diff
@@ -496,4 +496,47 @@ class ScaladocParserSuite extends FunSuite {
     )
   }
 
+  test("table") {
```

Comment:
Might be worth having test cases for parsing incomplete tables, for example missing pipes, multiple alignment rows, wrong characters in the alignment row, is it possible to escape a pipe character?

Reply:
will do.

* there's a already a test for escaped pipe char (line 504)
* the code already allows a different number of columns in each row
  * a row is essentially a line starting and ending with a pipe char
  * there's a test in `scalafmt` but will add it here as well
  * but doesn't allow a missing ending pipe char (will add a test)
* the code doesn't check that alignment characters are '-' or ':'; it simply checks if the cell starts or ends with ':'; should it be "more careful"?

Reply:
That sounds great! The biggest risk i can think of with formatting docstrings is that Scalafmt will ruin somebody’s ascii art or something else that should be left alone verbatim (even if it doesn’t render correctly in the Scalafix html).

Unrelated to this PR, is there a way for users to disable this feature for a specific docstring?

Reply:
> That sounds great! The biggest risk i can think of with formatting docstrings is that Scalafmt will ruin somebody’s ascii art or something else that should be left alone verbatim (even if it doesn’t render correctly in the Scalafix html).

We could parse out any lines which don't start with a letter as an opaque line, rather than splitting into text with words.

> Unrelated to this PR, is there a way for users to disable this feature for a specific docstring?

I am guessing `// format: off` should work, will test that in `scalafmt`.

Reply:
added a few more tests, as requested, and expanded the parser a bit.